lib/macho/macho_file.rb

```diff
@@ -264,6 +264,45 @@ def change_rpath(old_path, new_path)
       set_path_in_rpath(rpath_cmd, old_path, new_path)
     end
 
+    # Add the given runtime path to the Mach-O.
+    # @example
+    #  file.rpaths # => ["/lib"]
+    #  file.add_rpath("/usr/lib")
+    #  file.rpaths # => ["/lib", "/usr/lib"]
+    # @param path [String] the new runtime path
+    # @return [void]
+    # @raise [MachO::RpathExistsError] if the runtime path already exists
+    def add_rpath(path)
+      raise UnimplementedError.new("addition of rpaths")
+      raise RpathExistsError.new(path) if rpaths.include?(path)
+    end
+
+    # Delete the given runtime path from the Mach-O.
+    # @example
+    #  file.rpaths # => ["/lib"]
+    #  file.delete_rpath("/lib")
+    #  file.rpaths # => []
+    # @param path [String] the runtime path to delete
+    # @return void
+    # @raise [MachO::RpathUnknownError] if no such runtime path exists
+    def delete_rpath(path)
+      rpath_cmd = command(:LC_RPATH).find { |r| r.path.to_s == path }
```

Comment:
I just experimented with `install_name_tool` (from Xcode 7.3.1). Here's what I found out:

- `install_name_tool -add_rpath <duplicate>` reports an error if the given path already exists and refuses to add it.
- `install_name_tool -add_rpath <random>` followed by `install_name_tool -rpath <random> <duplicate>` doesn't complain and happily creates a duplicate entry.
- `install_name_tool -delete_rpath <duplicate>` only deletes the first of possibly several matching `LC_RPATH` commands.

I don't think we want to emulate all of those quirks and should be more strict, unless our ultimate aim is to be able to re-implement `install_name_tool` (which would also require silently ignoring errors for `-id` and `-change`).

I also quickly checked Mach-O binaries on 10.11.5. Of 3522 Mach-O files in `/System`, only 49 have an embedded `LC_RPATH` command (not surprising, as those binaries don't need to be relocatable) and there are no duplicate RPATHs (though there are some bogus and broken entries among them).